src/client/auth/oidc.rs

```diff
@@ -886,3 +916,90 @@ async fn send_sasl_command(
         response.auth_response_body(MONGODB_OIDC_STR)?,
     )
 }
+
+pub(super) fn validate_credential(credential: &Credential) -> Result<()> {
```

Comment:
This is mostly just copied over from its old location, with a few new validation cases to catch things we hadn't been that the newly synced spec tests showed up.